lib/rules/valid-next-tick.js

```diff
@@ -0,0 +1,167 @@
+/**
+ * @fileoverview enforce valid `nextTick` function calls
+ * @author Flo Edelmann
+ * @copyright 2021 Flo Edelmann. All rights reserved.
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const utils = require('../utils')
+const { findVariable } = require('eslint-utils')
+
+// ------------------------------------------------------------------------------
+// Helpers
+// ------------------------------------------------------------------------------
+
+/**
+ * @param {Identifier} identifier
+ * @param {RuleContext} context
+ * @returns {ASTNode|undefined}
+ */
+function getVueNextTickParentNode(identifier, context) {
+  // Instance API: this.$nextTick()
+  if (
+    identifier.name === '$nextTick' &&
+    identifier.parent.type === 'MemberExpression' &&
+    utils.isThis(identifier.parent.object, context)
+  ) {
+    return identifier.parent.parent
+  }
+
+  // Vue 2 Global API: Vue.nextTick()
+  if (
+    identifier.name === 'nextTick' &&
+    identifier.parent.type === 'MemberExpression' &&
+    identifier.parent.object.type === 'Identifier' &&
+    identifier.parent.object.name === 'Vue'
+  ) {
+    return identifier.parent.parent
+  }
+
+  // Vue 3 Global API: import { nextTick as nt } from 'vue'; nt()
+  const variable = findVariable(context.getScope(), identifier)
+
+  if (variable != null && variable.defs.length === 1) {
+    const def = variable.defs[0]
+    if (
+      def.type === 'ImportBinding' &&
+      def.node.type === 'ImportSpecifier' &&
+      def.node.imported.type === 'Identifier' &&
+      def.node.imported.name === 'nextTick' &&
+      def.node.parent.type === 'ImportDeclaration' &&
+      def.node.parent.source.value === 'vue'
+    ) {
+      return identifier.parent
+    }
+  }
+
+  return undefined
+}
+
+/**
+ * @param {CallExpression} callExpression
+ * @returns {boolean}
+ */
+function isAwaitedPromise(callExpression) {
+  return (
```

Comment:
Change it to check for `Promise.xxx()` calls such as `Promise.all()`.
You also probably need to consider `foo = nextTick()` to be asynchronous.

Reply:
Using `nextTick` only makes sense sequentially (e.g. after a value change, to wait for a DOM re-render). So using `Promise.all([Vue.nextTick, someOtherPromise])` is not really a common pattern and might indeed be a misuse of `nextTick`, right?

Reply:
Users may compare it to promises that are shorter than `nextTick()`. For example, `import(...)` is probably faster after the second time.
I don't think the following usage is misused.

```js
const [ , myUtility] = await Promise.all([nextTick(), import('./my-utility')])
myUtility.foo(vm)
```

Users may also use promises like queues.

```js
let queue =  nextTick()

if (foo) {
  queue = queue.then(() => procFoo())
}
if (bar) {
  queue = queue.then(() => procBar())
}

return queue
```